ui/lib/core/addon/components/confirm-action.js

```diff
@@ -14,44 +14,73 @@ import layout from '../templates/components/confirm-action';
- * @property {String} [cancelButtonText=Cancel] - The cancel button text.
+ * @param {Func} [onConfirmAction=null] - The action to take upon confirming.
+ * @param {String} [confirmTitle=Delete this?] - The title to display when confirming.
+ * @param {String} [confirmMessage=You will not be able to recover it later.] - The message to display when confirming.
+ * @param {String} [confirmButtonText=Delete] - The confirm button text.
+ * @param {String} [cancelButtonText=Cancel] - The cancel button text.
+ * @param {String} [buttonClasses] - A string to indicate the button class.
+ * @param {String} [horizontalPosition=auto-right] - For the position of the dropdown.
+ * @param {String} [verticalPosition=below] - For the position of the dropdown.
+ * @param {Boolean} [isRunning=false] - If action is still running disable the confirm.
+ * @param {Boolean} [disable=false] - To disable the confirm action.
  *
  */
 
-export default Component.extend({
-  layout,
-  tagName: '',
-  supportsDataTestProperties: true,
-  buttonText: 'Delete',
-  confirmTitle: 'Delete this?',
-  confirmMessage: 'You will not be able to recover it later.',
-  confirmButtonText: 'Delete',
-  cancelButtonText: 'Cancel',
-  horizontalPosition: 'auto-right',
-  verticalPosition: 'below',
-  isRunning: false,
-  disabled: false,
-  showConfirm: false,
-  onConfirmAction: null,
-
-  actions: {
-    toggleConfirm() {
-      this.toggleProperty('showConfirm');
-    },
-
-    onConfirm() {
-      const confirmAction = this.onConfirmAction;
-
-      if (typeof confirmAction !== 'function') {
-        throw new Error('confirm-action components expects `onConfirmAction` attr to be a function');
-      } else {
-        confirmAction();
-        this.toggleProperty('showConfirm');
-      }
-    },
-  },
-});
+export default class ConfirmActionComponent extends Component {
+  get horizontalPosition() {
+    return this.args.horizontalPosition || 'auto-right';
+  }
+
+  get verticalPosition() {
+    return this.args.verticalPosition || 'below';
+  }
+
+  get isRunning() {
+    return this.args.isRunning || false;
+  }
+
+  get disabled() {
+    return this.args.disabled || false;
+  }
+
+  get confirmTitle() {
+    return this.args.confirmTitle || 'Delete this?';
+  }
+
+  get confirmMessage() {
+    return this.args.confirmMessage || 'You will not be able to recover it later.';
+  }
+
+  get confirmButtonText() {
+    return this.args.confirmButtonText || 'Delete';
+  }
+
+  get cancelButtonText() {
+    return this.args.cancelButtonText || 'Cancel';
+  }
+
+  @action
+  closeButton(d) {
+    d.actions.close();
+  }
+
+  @action
+  toggleConfirm() {
+    // toggle
+    this.showConfirm = !this.showConfirm;
+  }
+
+  @action
+  onConfirm() {
+    const confirmAction = this.args.onConfirmAction;
+
+    if (typeof confirmAction !== 'function') {
+      throw new Error('confirm-action components expects `onConfirmAction` attr to be a function');
```

Comment:
I wonder if we could instead assert here (like we do in [pki adapter and other places](https://github.com/hashicorp/vault/blob/e5b1f718f1930d4033ed5d73469013426d505429/ui/app/adapters/pki.js#L8))

Reply:
Let me look into that!